src/scancode/resource.py

```diff
@@ -434,19 +436,40 @@ def create_resources(_seq, _top, _parent, _is_file):
         # not keep parents already walked and we walk topdown.
         parent_by_loc = {root.location: root}
 
+
+        def _walk():
```

Comment:
Can you articulate the benefit of having an inner closed function in this case?

Reply:
I created it with a depth parameter initially but kept it as an inner function after removing that parameter. I think it just makes it easier to think of this block of code that traverses through the directory. What do you think?

Reply:
> I think it just makes it easier to think of this block of code that traverses through the directory.

That's a valid point. There is still something that tickles me a bit somehow... the fact that this is closed on available variables and does not receive arguments makes it look a bit like magic and does not promote readability IMHO.... It feels a bit like some magic happened because of the closures. I personally try to avoid the implicit nature of closures when possible and only use them when they are actually needed (typically when that inner function is a higher order function that returns function(s).  

Based on all that, I am not sure this help with readability and clarity in this case and I would be more comfy with either:
1. do not use a function
2. OR use a function with explicit arguments and not closures (such as here https://github.com/nexB/scancode-toolkit/pull/1948/files#diff-a8e64273dc1cb3ca2a1ff8127a989c6cR401 )

Now for the sake of testability in unit tests, it could make sense to have that as a top level function instead... It may be the sanest and best way. And then you can test this a lot!

Reply:
Factoring the walk function out seems to be the best choice here. We could have a top-level `depth_walk` function and that would also allow us to reuse it in other places.